### PR TITLE
add helm/azure-resource-chart

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards https://github.com/giantswarm/giantswarm/issues/2441

### CODE REVIEW

File: helm/azure-resource-chart/values.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+clusterName: "test-cluster"
+commonDomain: "godsmack.westeurope.azure.gigantic.io"
+commonDomainResourceGroup: "godsmack"
+hostClusterCIDR: "10.0.0.0/16"
+hostClusterResourceGroup: "godsmack"
+hostClusterVirtualNetwork: "godsmack"
```
Comment (TheoBrigitte) [inline helm/azure-resource-chart/values.yaml]: Might want to remove those installation specific default values

File: helm/azure-operator-resource-chart/values.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+clusterName: "test-cluster"
+commonDomain: "godsmack.westeurope.azure.gigantic.io"
+commonDomainResourceGroup: "godsmack"
+hostClusterCIDR: "10.0.0.0/16"
+hostClusterResourceGroup: "godsmack"
+hostClusterVirtualNetwork: "godsmack"
```
Comment (TheoBrigitte) [inline helm/azure-operator-resource-chart/values.yaml]: Might want to remove those installation specific values

**Comment by TheoBrigitte** [comment]
Is a public quay repo ok for this ?

File: helm/azure-operator-resource-chart/templates/cluster.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      etcd:
+        name: "{{.Values.commonDomain}}"
+        resourceGroup: "{{.Values.commonDomainResourceGroup}}"
+      ingress:
+        name: "{{.Values.commonDomain}}"
+        resourceGroup: "{{.Values.commonDomainResourceGroup}}"
+    hostCluster:
+      cidr: "{{.Values.hostClusterCIDR}}"
+      resourceGroup: "{{.Values.hostClusterResourceGroup}}"
+      virtualNetwork: "{{.Values.hostClusterVirtualNetwork}}"
```
Comment (kopiczko) [inline helm/azure-operator-resource-chart/templates/cluster.yaml]: `hostCluster` is not used anymore.

File: helm/azure-operator-resource-chart/values.yaml (hunk 2 above)
Comment (kopiczko) [reply helm/azure-operator-resource-chart/values.yaml]: I think this is only maybe wrong in political sense.

**Comment by kopiczko** [comment]
You have to create the repo

File: helm/azure-operator-resource-chart/values.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+clusterName: "test-cluster"
+commonDomain: ""
+commonDomainResourceGroup: ""
```
Comment (kopiczko) [inline helm/azure-operator-resource-chart/values.yaml]: WDYT about putting here something like

```
commonDomain: "test-host-cluster.westeurope.azure.gigantic.io"
commonDomainResourceGroup: "test-host-cluster"
```

Then it's easy to just replace the value with the actual cluster name. The same thing you didi for `sshPublicKey`

**Comment by kopiczko** [comment]
@indyfree I never knew which way to go. Small fmt entry maybe, to spread the knowledge? Or you think it's enough to just nag ppl in the PRs?

**Comment by indyfree** [comment]
@kopiczko Your right. Altough nagging is fun I'm gonna get a `fmt` on the way :wink:

**Comment by TheoBrigitte** [comment]
While we are at nitpicking, what about quotes, do we usually prefer to have quotes around string ?

File: helm/azure-operator-resource-chart/values.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+clusterName: "test-cluster"
+commonDomain: "test-host-cluster.westeurope.azure.gigantic.io"
+commonDomainResourceGroup: "test-host-cluster"
+sshUser: "test-user"
+sshPublicKey: "ssh-rsa XXXXXXXXXXXXXXXXXXXXXXXXX"
```
Comment (rossf7) [inline helm/azure-operator-resource-chart/values.yaml]: We use this in aws-operator too but if it doesn't get overridden it causes problems.

As it's not valid base64 the cloudconfig fails. I think it's better to put a default key or just anything base64 here. So you can test the cluster comes up even if you can't SSH to the nodes.

File: helm/azure-operator-resource-chart/templates/cluster.yaml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        insecurePort: 8080
+        ip: "172.31.0.1"
+        securePort: 443
+      cloudProvider: "azure"
+      dns:
+        ip: "172.31.0.10"
+      domain: "cluster.local"
+      hyperkube:
+        docker:
+          image: "quay.io/giantswarm/hyperkube:v1.8.4_coreos.0"
```
Comment (rossf7) [inline helm/azure-operator-resource-chart/templates/cluster.yaml]: This docker image isn't used anymore. This is now set in k8scc so we can have the k8s version in the version bundles. Can we remove it?

File: helm/azure-operator-resource-chart/templates/cluster.yaml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+      ingressController:
+        docker:
+          image: "quay.io/giantswarm/nginx-ingress-controller:0.9.0-beta.11"
```
Comment (rossf7) [inline helm/azure-operator-resource-chart/templates/cluster.yaml]: Same for ingress controller.

**Comment by kopiczko** [comment]
> While we are at nitpicking, what about quotes, do we usually prefer to have quotes around string ?

I like them.

**Pull request merged**